# KMD009 - ald-defs-used

Attribute List Definitions must be referenced in the document.

**Tags:** kramdown, ald, attributes

**Aliases:** ald-defs-used

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown ALD definitions (`{:identifier: attrs}`) are only useful if referenced elsewhere with `{:identifier}`. Unused definitions add clutter.

## Examples

### Incorrect

```markdown
{:my-style: .highlight .bold}

No elements use this ALD.
```

### Correct

```markdown
{:my-style: .highlight .bold}

This paragraph uses the style.
{:my-style}
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD009 removes unused ALD definition lines.

## Related Rules

- [KMD006](kmd006.md) - IAL syntax must be well-formed

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
