# KMD008 - block-extension-syntax

Block extensions must be properly opened and closed.

**Tags:** kramdown, block-extensions

**Aliases:** block-extension-syntax

**Fixable:** Partial (unclosed openers only)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown block extensions (`{::name}` / `{:/name}`) must be properly balanced. Unclosed or mismatched extensions cause rendering errors.

## Examples

### Incorrect

```markdown
{::comment}
This comment is never closed.
```

### Correct

```markdown
{::comment}
This comment is properly closed.
{:/comment}
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD008 appends closing tags for unclosed block extensions. Mismatched or unexpected closing tags are not auto-fixable.

## Related Rules

None.

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
