# KMD003 - footnote-defs-used

Footnote definitions must be referenced in the document.

**Tags:** kramdown, footnotes

**Aliases:** footnote-defs-used

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Unused footnote definitions (`[^label]: text`) clutter the document. If no reference uses the footnote, the definition should be removed.

## Examples

### Incorrect

```markdown
Some content with no footnotes.

[^unused]: This footnote is never referenced.
```

### Correct

```markdown
Some content with a footnote[^1].

[^1]: This footnote is referenced.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD003 removes unused footnote definition lines.

## Related Rules

- [KMD002](kmd002.md) - Footnote references must have matching definitions

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
