# KMD006 - valid-ial-syntax

IAL (Inline Attribute List) syntax must be well-formed.

**Tags:** kramdown, ial, attributes

**Aliases:** valid-ial-syntax

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Block-level IALs (`{: #id .class key="value"}`) must follow Kramdown syntax rules. Malformed IALs are silently ignored by the renderer, hiding errors.

## Examples

### Incorrect

```markdown
{: invalid syntax here
```

### Correct

```markdown
{: #my-id .my-class key="value"}
{: .highlight}
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD006 removes malformed IAL lines.

## Related Rules

- [KMD009](kmd009.md) - ALD definitions must be referenced
- [KMD010](kmd010.md) - Inline IAL syntax must be well-formed

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
