# KMD011 - inline-math-balanced

Inline math spans must have balanced `$` delimiters.

**Tags:** kramdown, math

**Aliases:** inline-math-balanced

**Fixable:** No

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown uses `$` delimiters for inline math (`$E = mc^2$`). An odd number of unescaped `$` on a line indicates an unclosed math span.

## Examples

### Incorrect

```markdown
The formula $E = mc^2 is missing a closing delimiter.
```

### Correct

```markdown
The formula $E = mc^2$ is complete.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

This rule is not auto-fixable. Fixing unbalanced delimiters requires understanding where the closing `$` should be placed.

## Related Rules

- [KMD007](kmd007.md) - Math block delimiters
- [MD059](md059.md) - Emphasis markers and math syntax

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
