# KMD004 - abbreviation-defs-used

Abbreviation definitions should be used in document text.

**Tags:** kramdown, abbreviations

**Aliases:** abbreviation-defs-used

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown abbreviation definitions (`*[TERM]: expansion`) are only useful if the term appears in the document text. Unused definitions add clutter.

## Examples

### Incorrect

```markdown
No abbreviations used here.

*[HTML]: HyperText Markup Language
```

### Correct

```markdown
HTML is the standard markup language.

*[HTML]: HyperText Markup Language
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD004 removes unused abbreviation definition lines.

## Related Rules

None.

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
