# KMD005 - no-duplicate-heading-ids

Heading IDs must be unique within the document.

**Tags:** kramdown, headings, ids

**Aliases:** no-duplicate-heading-ids

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Heading IDs (auto-generated or explicit via `{#id}`) must be unique. Duplicate IDs cause ambiguous anchor links and break in-document navigation.

## Examples

### Incorrect

```markdown
## Setup

## Setup
```

Both generate the same ID `setup`.

### Correct

```markdown
## Setup

## Setup {#setup-2}
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD005 appends `{#slug-N}` to duplicate headings to make IDs unique.

## Related Rules

- [MD024](md024.md) - No duplicate heading content

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
