# KMD001 - definition-list-term-has-definition

Definition list terms must be followed by a definition.

**Tags:** kramdown, definition-lists

**Aliases:** definition-list-term-has-definition

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

In Kramdown, definition lists use a term followed by a `: definition` line. A term without a definition is likely an error.

## Examples

### Incorrect

```markdown
Apple
Banana

: A yellow fruit
```

"Apple" has no definition.

### Correct

```markdown
Apple
: A round fruit

Banana
: A yellow fruit
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD001 inserts a stub definition (`: `) after orphaned terms.

## Related Rules

None.

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
