# KMD007 - math-block-delimiters

Math block `$$` delimiters must be matched.

**Tags:** kramdown, math

**Aliases:** math-block-delimiters

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown math blocks use `$$` as opening and closing delimiters. An unclosed math block causes the rest of the document to be consumed as math content.

## Examples

### Incorrect

```markdown
$$
E = mc^2

Rest of document consumed as math...
```

Missing closing `$$`.

### Correct

```markdown
$$
E = mc^2
$$
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD007 appends a closing `$$` at the end of the file for unclosed math blocks.

## Related Rules

- [KMD011](kmd011.md) - Inline math balanced
- [MD059](md059.md) - Emphasis marker style and math conflicts

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
