# KMD010 - inline-ial-syntax

Inline IAL syntax must be well-formed.

**Tags:** kramdown, ial, attributes

**Aliases:** inline-ial-syntax

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Inline IALs (`*em*{: .class}`) follow the same attribute syntax as block IALs but apply to inline elements. Malformed inline IALs are silently ignored.

## Examples

### Incorrect

```markdown
*emphasis*{: invalid here}
```

### Correct

```markdown
*emphasis*{: .highlight}
*emphasis*{: #my-id .class key="value"}
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD010 removes malformed inline IAL tokens.

## Related Rules

- [KMD006](kmd006.md) - Block-level IAL syntax

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
