# KMD002 - footnote-refs-defined

Footnote references must have matching definitions.

**Tags:** kramdown, footnotes

**Aliases:** footnote-refs-defined

**Fixable:** Yes (auto-fix available)

**Enabled by default:** No (enable via kramdown preset)

## Rationale

Kramdown footnote references (`[^label]`) must have a corresponding definition (`[^label]: text`). Undefined footnotes result in broken references in the rendered output.

## Examples

### Incorrect

```markdown
This has a footnote[^1] but no definition.
```

### Correct

```markdown
This has a footnote[^1] with a definition.

[^1]: This is the footnote text.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, KMD002 appends stub definitions for undefined footnote labels at the end of the document.

## Related Rules

- [KMD003](kmd003.md) - Footnote definitions must be referenced

## Additional Information

This rule is specific to mkdlint's Kramdown extension and has no upstream equivalent.
