# MD007 - ul-indent

Unordered list indentation should be consistent.

**Tags:** bullet, ul, indentation

**Aliases:** ul-indent

**Fixable:** Yes (auto-fix available)

## Rationale

Nested unordered list items should be indented by a consistent number of spaces at each level. Inconsistent indentation makes lists harder to read and may render differently across Markdown parsers.

## Examples

### Incorrect

```markdown
* Item 1
   * Nested item (3 spaces - not a multiple of 2)
```

### Correct

```markdown
* Item 1
  * Nested item (2 spaces)
    * Deeply nested (4 spaces)
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `indent` | integer | `2` | Number of spaces per indentation level |

```json
{
  "MD007": {
    "indent": 4
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD007 adjusts the leading whitespace of nested list items to be the nearest correct multiple of the configured indent value.

## Related Rules

- [MD004](md004.md) - Unordered list marker style
- [MD005](md005.md) - Inconsistent indentation for list items
- [MD030](md030.md) - Spaces after list markers

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md007.md)
