# MD004 - ul-style

Unordered list style should be consistent.

**Tags:** bullet, ul

**Aliases:** ul-style

**Fixable:** Yes (auto-fix available)

## Rationale

Mixing different unordered list markers (`*`, `-`, `+`) within a document creates visual inconsistency. Enforcing a single style improves readability.

## Examples

### Incorrect

```markdown
* Item 1
- Item 2
+ Item 3
```

### Correct

```markdown
- Item 1
- Item 2
- Item 3
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent`, `asterisk`, `dash`, `plus`, `sublist` |

```json
{
  "MD004": {
    "style": "dash"
  }
}
```

**Style descriptions:**

- `consistent` — First list marker determines the style for the rest
- `asterisk` — All markers must be `*`
- `dash` — All markers must be `-`
- `plus` — All markers must be `+`
- `sublist` — Sublists must use a different marker than their parent list

## Auto-fix Behavior

When `--fix` is used, MD004 replaces incorrect list markers with the expected one.

## Related Rules

- [MD005](md005.md) - List indentation consistency
- [MD007](md007.md) - Unordered list indentation
- [MD032](md032.md) - Lists should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md004.md)
