# MD001 - heading-increment

Heading levels should only increment by one level at a time.

**Tags:** headings

**Aliases:** heading-increment, header-increment

**Fixable:** No auto-fix

## Rationale

Markdown headings create a document outline. Skipping heading levels (e.g., going from H1 directly to H3) can confuse readers and break document structure in tools that generate tables of contents.

## Examples

### Incorrect

```markdown
# Heading 1

### Heading 3

We skipped heading level 2\!
```

This violates MD001 because the H3 follows an H1 without an H2 in between.

### Correct

```markdown
# Heading 1

## Heading 2

### Heading 3

Proper heading hierarchy.
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `front_matter_title` | string or `false` | `"^\s*\"?title\"?\s*[:=]"` | Regex to detect title in front matter. Set to `false` to disable. |

If front matter contains a title field matching the pattern, MD001 treats it as an implicit H1:

```markdown
---
title: Document Title
---

## Section 1

This is valid - the front matter title acts as H1.
```

## Auto-fix Behavior

This rule does not provide auto-fix. Fixing heading level issues would require knowing the intended document structure.

## Related Rules

- [MD025](md025.md) - Ensures only one top-level heading
- [MD041](md041.md) - Requires first line to be a top-level heading

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md001.md)
