# MD003 - heading-style

Heading style should be consistent throughout the document.

**Tags:** headings

**Aliases:** heading-style

**Fixable:** Yes (auto-fix available)

## Rationale

Mixing heading styles (ATX, setext, closed ATX) within a single document creates visual inconsistency and can confuse readers.

## Examples

### Incorrect

```markdown
# ATX Heading

Setext Heading
--------------
```

Mixing ATX and setext styles in the same document.

### Correct (ATX)

```markdown
# Heading 1

## Heading 2

### Heading 3
```

### Correct (Setext with ATX for H3+)

```markdown
Heading 1
=========

Heading 2
---------

### Heading 3
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent`, `atx`, `atx_closed`, `setext`, `setext_with_atx`, `setext_with_atx_closed` |

```json
{
  "MD003": {
    "style": "atx"
  }
}
```

**Style descriptions:**

- `consistent` — First heading determines the style for the rest of the document
- `atx` — `# Heading`
- `atx_closed` — `# Heading #`
- `setext` — Underlined with `=` (H1) or `-` (H2); H3-H6 fall back to ATX
- `setext_with_atx` — Setext for H1/H2, ATX for H3-H6
- `setext_with_atx_closed` — Setext for H1/H2, closed ATX for H3-H6

## Auto-fix Behavior

When `--fix` is used, MD003 converts headings to match the expected style. When converting from setext to ATX, the underline is also removed.

## Related Rules

- [MD022](md022.md) - Headings should be surrounded by blank lines
- [MD023](md023.md) - Headings must start at the beginning of the line

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md003.md)
