# MD009 - no-trailing-spaces

Trailing spaces on lines.

**Tags:** whitespace

**Aliases:** no-trailing-spaces

**Fixable:** Yes (auto-fix available)

## Rationale

Trailing whitespace is invisible, adds noise to diffs, and serves no purpose in Markdown. Some editors automatically strip trailing whitespace, which can cause unexpected diff changes.

## Examples

### Incorrect

```markdown
Some text with trailing spaces   
Another line with a tab	
```

### Correct

```markdown
Some text without trailing spaces
Another clean line
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD009 removes all trailing whitespace (spaces and tabs) from every line.

## Related Rules

- [MD010](md010.md) - Hard tabs
- [MD012](md012.md) - Multiple consecutive blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md009.md)
