# MD018 - no-missing-space-atx

No space after hash on atx style heading.

**Tags:** headings, atx, spaces

**Aliases:** no-missing-space-atx

**Fixable:** Yes (auto-fix available)

## Rationale

ATX headings require a space between the hash marks and the heading text. Without the space, some parsers may not recognize the line as a heading.

## Examples

### Incorrect

```markdown
#Heading without space
##Another heading
```

### Correct

```markdown
# Heading with space
## Another heading
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD018 inserts a space after the hash marks.

## Related Rules

- [MD019](md019.md) - Multiple spaces after hash on atx style heading
- [MD020](md020.md) - No space inside hashes on closed atx style heading
- [MD003](md003.md) - Heading style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md018.md)
