# MD014 - commands-show-output

Dollar signs used before commands without showing output.

**Tags:** code

**Aliases:** commands-show-output

**Fixable:** Yes (auto-fix available)

## Rationale

When documenting shell commands in code blocks, `$` prefixes are only useful when mixed with output to distinguish commands from their results. If all lines start with `$`, the prefixes are redundant and make it harder to copy-paste commands.

## Examples

### Incorrect

```markdown
    ```bash
    $ npm install
    $ npm test
    ```
```

All lines have `$` prefixes but no output is shown.

### Correct (no dollar signs)

```markdown
    ```bash
    npm install
    npm test
    ```
```

### Correct (mixed commands and output)

```markdown
    ```bash
    $ npm test
    All tests passed\!
    ```
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD014 removes the `$ ` prefix (or just `$` if no space follows) from command lines in code blocks.

## Related Rules

- [MD060](md060.md) - Dollar signs in fenced code blocks

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md014.md)
