# MD019 - no-multiple-space-atx

Multiple spaces after hash on atx style heading.

**Tags:** headings, atx, spaces

**Aliases:** no-multiple-space-atx

**Fixable:** Yes (auto-fix available)

## Rationale

ATX headings should have exactly one space between the hash marks and the heading text. Multiple spaces create inconsistency.

## Examples

### Incorrect

```markdown
#  Heading with two spaces
##   Another heading with three spaces
```

### Correct

```markdown
# Heading with one space
## Another heading
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD019 collapses multiple spaces after hash marks to a single space.

## Related Rules

- [MD018](md018.md) - No space after hash on atx style heading
- [MD021](md021.md) - Multiple spaces inside hashes on closed atx style heading
- [MD003](md003.md) - Heading style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md019.md)
