# MD013 - line-length

Line length should not exceed a configured limit.

**Tags:** line_length

**Aliases:** line-length

**Fixable:** No

## Rationale

Long lines are harder to read in source form, especially in terminals, code reviews, and side-by-side diffs. A maximum line length encourages better formatting.

## Examples

### Incorrect

```markdown
This is a very long line that exceeds the default 80 character limit and should be broken into shorter lines for readability.
```

### Correct

```markdown
This is a shorter line that fits within the 80 character
limit and is easier to read in source form.
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `line_length` | integer | `80` | Maximum allowed line length in characters |

```json
{
  "MD013": {
    "line_length": 120
  }
}
```

**Excluded content:** Code blocks, tables, and headings are excluded from this check.

## Auto-fix Behavior

This rule is not auto-fixable. Line wrapping requires understanding the content structure and would need human judgment about where to break lines.

## Related Rules

- [MD009](md009.md) - Trailing spaces

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md)
