# MD010 - no-hard-tabs

Hard tabs should not be used.

**Tags:** whitespace, hard_tab

**Aliases:** no-hard-tabs

**Fixable:** Yes (auto-fix available)

## Rationale

Hard tab characters render with different widths depending on the editor or viewer, leading to inconsistent document appearance. Using spaces instead ensures consistent formatting everywhere.

## Examples

### Incorrect

```markdown
→Indented with a tab
Text→with embedded tab
```

(→ represents a tab character)

### Correct

```markdown
    Indented with spaces
Text    with spaces
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD010 replaces each hard tab character with 4 spaces.

## Related Rules

- [MD009](md009.md) - Trailing spaces

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md010.md)
