# MD011 - no-reversed-links

Reversed link syntax.

**Tags:** links

**Aliases:** no-reversed-links

**Fixable:** Yes (auto-fix available)

## Rationale

A common mistake is writing `(text)[url]` instead of the correct `[text](url)`. This results in broken links in the rendered output.

## Examples

### Incorrect

```markdown
(Click here)[https://example.com]
```

### Correct

```markdown
[Click here](https://example.com)
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD011 swaps the parentheses and brackets to produce the correct `[text](url)` syntax.

## Related Rules

- [MD034](md034.md) - Bare URLs
- [MD039](md039.md) - Spaces inside link text
- [MD042](md042.md) - No empty links

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md011.md)
