# MD012 - no-multiple-blanks

Multiple consecutive blank lines.

**Tags:** whitespace, blank_lines

**Aliases:** no-multiple-blanks

**Fixable:** Yes (auto-fix available)

## Rationale

Multiple consecutive blank lines do not add extra vertical space in rendered Markdown. They only add visual clutter to the source file.

## Examples

### Incorrect

```markdown
# Heading


Content after two blank lines
```

### Correct

```markdown
# Heading

Content after one blank line
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD012 removes extra blank lines, leaving at most one consecutive blank line.

## Related Rules

- [MD022](md022.md) - Headings should be surrounded by blank lines
- [MD047](md047.md) - Files should end with a single newline

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md012.md)
