# MD024 - no-duplicate-heading

Multiple headings with the same content.

**Tags:** headings

**Aliases:** no-duplicate-heading, no-duplicate-header

**Fixable:** No

## Rationale

Duplicate heading content can confuse readers and create ambiguous anchor links. Each heading should be unique to enable reliable cross-references.

## Examples

### Incorrect

```markdown
# Overview

## Details

# Overview
```

Two headings with identical text "Overview".

### Correct

```markdown
# Project Overview

## Details

# API Overview
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

This rule is not auto-fixable. Renaming headings requires understanding the document context and intent.

## Related Rules

- [MD025](md025.md) - Multiple top-level headings
- [KMD005](kmd005.md) - No duplicate heading IDs

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md)
