# MD028 - no-blanks-blockquote

Blank line inside blockquote.

**Tags:** blockquote, whitespace

**Aliases:** no-blanks-blockquote

**Fixable:** Yes (auto-fix available)

## Rationale

A blank line within a blockquote ends the blockquote in most parsers. If you want a blank line within a blockquote, use `>` on the blank line to continue the blockquote.

## Examples

### Incorrect

```markdown
> First paragraph

> This is actually a new blockquote, not a continuation
```

### Correct

```markdown
> First paragraph
>
> Second paragraph (still in the same blockquote)
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD028 adds `>` to blank lines between blockquote paragraphs to maintain a single continuous blockquote.

## Related Rules

- [MD027](md027.md) - Multiple spaces after blockquote symbol

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md028.md)
