# MD023 - heading-start-left

Headings must start at the beginning of the line.

**Tags:** headings, spaces

**Aliases:** heading-start-left

**Fixable:** Yes (auto-fix available)

## Rationale

Headings should start at column 1 (no leading whitespace). Indented headings may not be recognized by some parsers or may be visually confusing.

## Examples

### Incorrect

```markdown
  # Indented heading
    ## Another indented heading
```

### Correct

```markdown
# Heading at column 1
## Another heading at column 1
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD023 removes the leading whitespace before headings.

## Related Rules

- [MD022](md022.md) - Headings should be surrounded by blank lines
- [MD003](md003.md) - Heading style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md023.md)
