# MD021 - no-multiple-space-closed-atx

Multiple spaces inside hashes on closed atx style heading.

**Tags:** headings, atx, atx_closed, spaces

**Aliases:** no-multiple-space-closed-atx

**Fixable:** Yes (auto-fix available)

## Rationale

Closed ATX headings should have exactly one space between the hash marks and the heading text on both sides for consistency.

## Examples

### Incorrect

```markdown
#  Heading  #
##  Heading  ##
```

### Correct

```markdown
# Heading #
## Heading ##
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD021 collapses multiple spaces inside closed ATX headings to a single space on each side.

## Related Rules

- [MD020](md020.md) - No space inside hashes on closed atx style heading
- [MD019](md019.md) - Multiple spaces after hash on atx style heading

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md021.md)
