# MD020 - no-missing-space-closed-atx

No space inside hashes on closed atx style heading.

**Tags:** headings, atx, atx_closed, spaces

**Aliases:** no-missing-space-closed-atx

**Fixable:** Yes (auto-fix available)

## Rationale

Closed ATX headings (`# Heading #`) require spaces between the hash marks and the heading text on both sides for proper parsing and readability.

## Examples

### Incorrect

```markdown
#Heading#
## Heading##
```

### Correct

```markdown
# Heading #
## Heading ##
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD020 inserts spaces where missing inside closed ATX headings.

## Related Rules

- [MD021](md021.md) - Multiple spaces inside hashes on closed atx style heading
- [MD018](md018.md) - No space after hash on atx style heading

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md020.md)
