# MD025 - single-title

Multiple top-level headings in the same document.

**Tags:** headings

**Aliases:** single-title, single-h1

**Fixable:** No

## Rationale

A document should have a single top-level (H1) heading that serves as the document title. Multiple H1 headings suggest the document should either be split or have its structure reorganized.

## Examples

### Incorrect

```markdown
# Title

Content here.

# Another Title
```

### Correct

```markdown
# Title

## Section 1

## Section 2
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

This rule is not auto-fixable. Resolving multiple H1 headings requires understanding the document structure.

## Related Rules

- [MD001](md001.md) - Heading increment
- [MD041](md041.md) - First line should be a top-level heading

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md025.md)
