# MD029 - ol-prefix

Ordered list item prefix.

**Tags:** ol

**Aliases:** ol-prefix

**Fixable:** Yes (auto-fix available)

## Rationale

Ordered lists can use different numbering styles. Enforcing a consistent style improves readability and makes reordering easier.

## Examples

### Incorrect (with style `one`)

```markdown
1. First
2. Second
3. Third
```

### Correct (with style `one`)

```markdown
1. First
1. Second
1. Third
```

### Correct (with style `ordered`)

```markdown
1. First
2. Second
3. Third
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"one_or_ordered"` | `one`, `ordered`, `zero`, `one_or_ordered` |

```json
{
  "MD029": {
    "style": "ordered"
  }
}
```

**Style descriptions:**

- `one` — All items must use `1.`
- `ordered` — Items must use incrementing numbers (`1.`, `2.`, `3.`, ...)
- `zero` — All items must use `0.`
- `one_or_ordered` — Either all `1.` or incrementing numbers (default)

## Auto-fix Behavior

When `--fix` is used, MD029 renumbers ordered list items to match the configured style.

## Related Rules

- [MD030](md030.md) - Spaces after list markers
- [MD032](md032.md) - Lists should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md029.md)
