# MD026 - no-trailing-punctuation

Trailing punctuation in heading.

**Tags:** headings

**Aliases:** no-trailing-punctuation

**Fixable:** Yes (auto-fix available)

## Rationale

Headings should not end with punctuation characters like periods, commas, or semicolons. These are typically unnecessary and look inconsistent in tables of contents.

## Examples

### Incorrect

```markdown
# This is a heading.
## Introduction:
```

### Correct

```markdown
# This is a heading
## Introduction
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD026 removes the trailing punctuation character from headings.

## Related Rules

- [MD003](md003.md) - Heading style
- [MD022](md022.md) - Headings should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md026.md)
