# MD022 - blanks-around-headings

Headings should be surrounded by blank lines.

**Tags:** headings, blank_lines

**Aliases:** blanks-around-headings, blanks-around-headers

**Fixable:** Yes (auto-fix available)

## Rationale

Blank lines before and after headings improve readability and ensure consistent rendering across different Markdown parsers. Some parsers require blank lines around headings for proper detection.

## Examples

### Incorrect

```markdown
Some text
# Heading
More text
```

### Correct

```markdown
Some text

# Heading

More text
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD022 inserts blank lines before and/or after headings where missing.

## Related Rules

- [MD003](md003.md) - Heading style
- [MD023](md023.md) - Headings must start at the beginning of the line
- [MD031](md031.md) - Fenced code blocks should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md022.md)
