# MD027 - no-multiple-space-blockquote

Multiple spaces after blockquote symbol.

**Tags:** blockquote, whitespace, indentation

**Aliases:** no-multiple-space-blockquote

**Fixable:** Yes (auto-fix available)

## Rationale

Blockquotes should have exactly one space after the `>` symbol. Multiple spaces create inconsistency and can cause unexpected indentation in rendered output.

## Examples

### Incorrect

```markdown
>  Double space after blockquote
>   Triple space after blockquote
```

### Correct

```markdown
> Single space after blockquote
> Another properly formatted line
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD027 collapses multiple spaces after `>` to a single space.

## Related Rules

- [MD028](md028.md) - Blank line inside blockquote

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md027.md)
