# MD030 - list-marker-space

Spaces after list markers.

**Tags:** whitespace, ol, ul

**Aliases:** list-marker-space

**Fixable:** Yes (auto-fix available)

## Rationale

The number of spaces after a list marker should be consistent. This improves readability and ensures proper nesting across different parsers.

## Examples

### Incorrect

```markdown
-  Two spaces after dash
*   Three spaces after asterisk
1.   Three spaces after number
```

### Correct

```markdown
- One space after dash
* One space after asterisk
1. One space after number
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `ul_single` | integer | `1` | Spaces after UL marker for single-line items |
| `ol_single` | integer | `1` | Spaces after OL marker for single-line items |
| `ul_multi` | integer | `1` | Spaces after UL marker for multi-line items |
| `ol_multi` | integer | `1` | Spaces after OL marker for multi-line items |

```json
{
  "MD030": {
    "ul_single": 1,
    "ol_single": 1,
    "ul_multi": 1,
    "ol_multi": 1
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD030 adjusts the number of spaces after list markers to match the configured values.

## Related Rules

- [MD004](md004.md) - Unordered list style
- [MD005](md005.md) - List indentation
- [MD029](md029.md) - Ordered list item prefix

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md030.md)
