# MD038 - no-space-in-code

Spaces inside code span elements.

**Tags:** whitespace, code

**Aliases:** no-space-in-code

**Fixable:** Yes (auto-fix available)

## Rationale

Spaces immediately inside code span backticks (`` ` code ` `` instead of `` `code` ``) create unnecessary whitespace in rendered output.

## Examples

### Incorrect

```markdown
This is ` code ` with spaces.
```

### Correct

```markdown
This is `code` without spaces.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD038 removes spaces immediately inside code span backticks.

## Related Rules

- [MD037](md037.md) - Spaces inside emphasis markers
- [MD040](md040.md) - Fenced code blocks should have a language specified

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md038.md)
