# MD032 - blanks-around-lists

Lists should be surrounded by blank lines.

**Tags:** bullet, ul, ol, blank_lines

**Aliases:** blanks-around-lists

**Fixable:** Yes (auto-fix available)

## Rationale

Blank lines before and after lists improve readability and ensure proper rendering. Without blank lines, list items may not be parsed correctly or may merge with surrounding content.

## Examples

### Incorrect

```markdown
Some text
- Item 1
- Item 2
More text
```

### Correct

```markdown
Some text

- Item 1
- Item 2

More text
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD032 inserts blank lines before and/or after lists where missing.

## Related Rules

- [MD022](md022.md) - Headings should be surrounded by blank lines
- [MD031](md031.md) - Fenced code blocks should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md032.md)
