# MD034 - no-bare-urls

Bare URL used.

**Tags:** links, url

**Aliases:** no-bare-urls

**Fixable:** Yes (auto-fix available)

## Rationale

Bare URLs should be wrapped in angle brackets (`<url>`) or formatted as proper links (`[text](url)`) for consistent rendering and accessibility.

## Examples

### Incorrect

```markdown
Visit https://example.com for more info.
```

### Correct

```markdown
Visit <https://example.com> for more info.
Visit [Example](https://example.com) for more info.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD034 wraps bare URLs in angle brackets (e.g., `https://example.com` becomes `<https://example.com>`).

## Related Rules

- [MD011](md011.md) - Reversed link syntax
- [MD039](md039.md) - Spaces inside link text

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md034.md)
