# MD031 - blanks-around-fences

Fenced code blocks should be surrounded by blank lines.

**Tags:** code, blank_lines

**Aliases:** blanks-around-fences

**Fixable:** Yes (auto-fix available)

## Rationale

Blank lines before and after fenced code blocks improve readability and ensure proper rendering across parsers. Without blank lines, the code fence may not be recognized.

## Examples

### Incorrect

```markdown
Some text
```code
code here
```
More text
```

### Correct

```markdown
Some text

```code
code here
```

More text
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `list_items` | boolean | `true` | Whether to check fenced code blocks inside list items |

```json
{
  "MD031": {
    "list_items": true
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD031 inserts blank lines before and/or after fenced code blocks where missing.

## Related Rules

- [MD022](md022.md) - Headings should be surrounded by blank lines
- [MD032](md032.md) - Lists should be surrounded by blank lines
- [MD040](md040.md) - Fenced code blocks should have a language specified

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md031.md)
