# MD035 - hr-style

Horizontal rule style should be consistent.

**Tags:** hr

**Aliases:** hr-style

**Fixable:** Yes (auto-fix available)

## Rationale

Horizontal rules can be created with different characters and quantities (`---`, `***`, `___`, etc.). Using a consistent style throughout the document improves readability.

## Examples

### Incorrect

```markdown
---

Content

***

More content

___
```

### Correct

```markdown
---

Content

---

More content

---
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent` or any literal HR string (e.g., `"---"`, `"***"`, `"___"`) |

```json
{
  "MD035": {
    "style": "---"
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD035 replaces inconsistent horizontal rules with the expected style.

## Related Rules

None.

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md035.md)
