# MD036 - no-emphasis-as-heading

Emphasis used instead of a heading.

**Tags:** headings, emphasis

**Aliases:** no-emphasis-as-heading

**Fixable:** Yes (auto-fix available)

## Rationale

Using bold or italic text as a pseudo-heading (a standalone emphasized line) creates a heading that is invisible to the document outline and accessibility tools. Use proper headings instead.

## Examples

### Incorrect

```markdown
**This looks like a heading**

Some content follows.
```

### Correct

```markdown
## This is a proper heading

Some content follows.
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `punctuation` | string | `".,;:!?;"` | Punctuation characters at line end that disqualify emphasis-as-heading detection |

Lines ending with these characters are not considered pseudo-headings (they're likely emphasized sentences, not headings).

## Auto-fix Behavior

When `--fix` is used, MD036 converts the emphasized line to a proper ATX heading.

## Related Rules

- [MD003](md003.md) - Heading style
- [MD022](md022.md) - Headings should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md036.md)
