# MD033 - no-inline-html

Inline HTML should not be used.

**Tags:** html

**Aliases:** no-inline-html

**Fixable:** No

## Rationale

Raw HTML in Markdown breaks the abstraction of the Markdown format. Documents with inline HTML may not render correctly in all contexts (e.g., some documentation systems strip HTML for security).

## Examples

### Incorrect

```markdown
<b>Bold text</b>
<br>
<div class="special">Content</div>
```

### Correct

```markdown
**Bold text**

Content in standard Markdown
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `allowed_elements` | string array | `[]` | HTML elements that are allowed |
| `table_allowed_elements` | string array | `[]` | HTML elements allowed inside tables |

```json
{
  "MD033": {
    "allowed_elements": ["br", "img"],
    "table_allowed_elements": ["br"]
  }
}
```

## Auto-fix Behavior

This rule is not auto-fixable. Replacing HTML with Markdown equivalents requires understanding the document semantics.

## Related Rules

- [MD045](md045.md) - Images should have alternate text

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md033.md)
