# MD037 - no-space-in-emphasis

Spaces inside emphasis markers.

**Tags:** whitespace, emphasis

**Aliases:** no-space-in-emphasis

**Fixable:** Yes (auto-fix available)

## Rationale

Spaces immediately inside emphasis markers (`* text *` instead of `*text*`) may not render correctly in all parsers and create visual inconsistency.

## Examples

### Incorrect

```markdown
This is * not emphasized * text.
This is ** not bold ** text.
```

### Correct

```markdown
This is *emphasized* text.
This is **bold** text.
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD037 removes spaces immediately inside emphasis markers.

## Related Rules

- [MD049](md049.md) - Emphasis style
- [MD050](md050.md) - Strong style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md037.md)
