# MD039 - no-space-in-links

Spaces inside link text.

**Tags:** whitespace, links

**Aliases:** no-space-in-links

**Fixable:** Yes (auto-fix available)

## Rationale

Spaces immediately inside link text brackets (`[ text ](url)` instead of `[text](url)`) create unnecessary whitespace and visual inconsistency.

## Examples

### Incorrect

```markdown
[ Click here ](https://example.com)
```

### Correct

```markdown
[Click here](https://example.com)
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD039 removes spaces immediately inside link text brackets.

## Related Rules

- [MD011](md011.md) - Reversed link syntax
- [MD042](md042.md) - No empty links

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md039.md)
