# MD042 - no-empty-links

No empty links.

**Tags:** links

**Aliases:** no-empty-links

**Fixable:** Partial (inline links only)

## Rationale

Links with empty URLs (`[text]()` or `[text](#)`) are broken and provide no value. They often indicate incomplete work or copy-paste errors.

## Examples

### Incorrect

```markdown
[Click here]()
[Empty fragment](#)
```

### Correct

```markdown
[Click here](https://example.com)
[Jump to section](#section-name)
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD042 provides fixes for empty inline links only. Reference-style empty links are not auto-fixable.

## Related Rules

- [MD011](md011.md) - Reversed link syntax
- [MD039](md039.md) - Spaces inside link text
- [MD051](md051.md) - Link fragments should be valid

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md042.md)
