# MD046 - code-block-style

Code block style should be consistent.

**Tags:** code

**Aliases:** code-block-style

**Fixable:** Yes (auto-fix available)

## Rationale

Markdown supports both fenced code blocks (````) and indented code blocks (4 spaces). Mixing styles within a document creates inconsistency.

## Examples

### Incorrect (mixed styles)

```markdown
    indented code block

```
fenced code block
```
```

### Correct (all fenced)

```markdown
```
first code block
```

```
second code block
```
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent`, `fenced`, `indented` |

```json
{
  "MD046": {
    "style": "fenced"
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD046 converts code blocks to the expected style. When converting from indented to fenced, appropriate fence markers are added.

## Related Rules

- [MD040](md040.md) - Fenced code blocks should have a language specified
- [MD048](md048.md) - Code fence style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md046.md)
