# MD043 - required-headings

Required heading structure.

**Tags:** headings

**Aliases:** required-headings, required-headers

**Fixable:** No

## Rationale

Some projects require a specific heading structure (e.g., every document must have "Overview", "Usage", and "License" sections). This rule enforces that required headings appear in the expected order.

## Examples

### Incorrect (with required headings `["# Title", "## Overview", "## Usage"]`)

```markdown
# Title

## Usage
```

Missing the required "## Overview" heading.

### Correct

```markdown
# Title

## Overview

Content here.

## Usage

More content.
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `headings` | string array | `[]` | Required heading patterns in order |

```json
{
  "MD043": {
    "headings": ["# Title", "## Overview", "## Usage"]
  }
}
```

Use `*` as a wildcard to allow any heading at that position.

## Auto-fix Behavior

This rule is not auto-fixable. Adding missing headings requires human judgment about content placement.

## Related Rules

- [MD024](md024.md) - No duplicate headings
- [MD025](md025.md) - Single top-level heading

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md043.md)
