# MD040 - fenced-code-language

Fenced code blocks should have a language specified.

**Tags:** code, language

**Aliases:** fenced-code-language

**Fixable:** Yes (auto-fix available)

## Rationale

Specifying a language on fenced code blocks enables syntax highlighting and helps readers understand the code content.

## Examples

### Incorrect

```markdown
```
const x = 1;
```
```

### Correct

```markdown
```javascript
const x = 1;
```
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `default_language` | string | `"text"` | Language to insert when a code block has none |

```json
{
  "MD040": {
    "default_language": "text"
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD040 inserts the configured default language after the opening fence.

## Related Rules

- [MD031](md031.md) - Fenced code blocks should be surrounded by blank lines
- [MD046](md046.md) - Code block style
- [MD048](md048.md) - Code fence style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md040.md)
