# MD041 - first-line-heading

First line in a file should be a top-level heading.

**Tags:** headings

**Aliases:** first-line-heading, first-line-h1

**Fixable:** Yes (auto-fix available)

## Rationale

Documents should begin with a top-level heading (H1) to establish the document title. This helps readers, accessibility tools, and table-of-contents generators.

## Examples

### Incorrect

```markdown
Some introductory text.

# Title
```

### Correct

```markdown
# Title

Some introductory text.
```

## Configuration

No configuration options for this rule.

Note: Front matter (e.g., YAML `---` blocks) is skipped when determining the first line.

## Auto-fix Behavior

When `--fix` is used, MD041 inserts a `# ` heading at the beginning of the file.

## Related Rules

- [MD001](md001.md) - Heading increment
- [MD025](md025.md) - Multiple top-level headings

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md041.md)
