# MD048 - code-fence-style

Code fence style should be consistent.

**Tags:** code

**Aliases:** code-fence-style

**Fixable:** Yes (auto-fix available)

## Rationale

Fenced code blocks can use backticks or tildes. Using a consistent delimiter throughout the document improves readability.

## Examples

### Incorrect

```markdown
```javascript
const x = 1;
```

~~~python
y = 2
~~~
```

### Correct

```markdown
```javascript
const x = 1;
```

```python
y = 2
```
```

## Configuration

No configuration options for this rule. The first fence delimiter determines the expected style in `consistent` mode.

## Auto-fix Behavior

When `--fix` is used, MD048 converts all fence delimiters to match the first one used in the document.

## Related Rules

- [MD040](md040.md) - Fenced code blocks should have a language specified
- [MD046](md046.md) - Code block style

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md048.md)
