# MD044 - proper-names

Proper names should have the correct capitalization.

**Tags:** spelling

**Aliases:** proper-names

**Fixable:** Yes (auto-fix available)

## Rationale

Proper names (brands, technologies, etc.) have specific capitalization. Incorrect capitalization (e.g., "github" instead of "GitHub") looks unprofessional.

## Examples

### Incorrect

```markdown
We use javascript and github for our project.
```

### Correct

```markdown
We use JavaScript and GitHub for our project.
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `names` | string array | `[]` | Proper names to enforce (case-sensitive) |
| `code_blocks` | boolean | `true` | Whether to check inside code blocks |

```json
{
  "MD044": {
    "names": ["JavaScript", "GitHub", "TypeScript"],
    "code_blocks": false
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD044 replaces incorrectly capitalized names with the correct form.

## Related Rules

None.

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md044.md)
