# MD045 - no-alt-text

Images should have alternate text (alt text).

**Tags:** accessibility, images

**Aliases:** no-alt-text

**Fixable:** Yes (auto-fix available)

## Rationale

Alt text is essential for accessibility. Screen readers use alt text to describe images to visually impaired users. Images without alt text are inaccessible.

## Examples

### Incorrect

```markdown
![](image.png)
```

### Correct

```markdown
![A descriptive alt text](image.png)
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD045 inserts a placeholder alt text for images missing it.

## Related Rules

- [MD033](md033.md) - No inline HTML (HTML img tags should also have alt)

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md045.md)
