# MD049 - emphasis-style

Emphasis style should be consistent.

**Tags:** emphasis

**Aliases:** emphasis-style

**Fixable:** Yes (auto-fix available)

## Rationale

Emphasis can use asterisks (`*text*`) or underscores (`_text_`). Using a consistent style throughout the document improves readability.

## Examples

### Incorrect

```markdown
This is *italic* and this is _also italic_.
```

### Correct

```markdown
This is *italic* and this is *also italic*.
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent`, `asterisk`, `underscore` |

```json
{
  "MD049": {
    "style": "asterisk"
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD049 converts emphasis markers to match the expected style.

## Related Rules

- [MD050](md050.md) - Strong style
- [MD037](md037.md) - Spaces inside emphasis markers

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md049.md)
