# MD054 - link-image-style

Link and image style should be consistent.

**Tags:** links, images

**Aliases:** link-image-style

**Fixable:** Partial

## Rationale

Markdown supports multiple link/image styles: autolink, inline, full reference, collapsed reference, and shortcut reference. Restricting which styles are allowed enforces consistency.

## Examples

### Incorrect (if shortcut references are disallowed)

```markdown
[Click here]

[Click here]: https://example.com
```

### Correct (using inline style)

```markdown
[Click here](https://example.com)
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `autolink` | boolean | `true` | Allow autolink style (`<url>`) |
| `inline` | boolean | `true` | Allow inline style (`[text](url)`) |
| `full` | boolean | `true` | Allow full reference (`[text][label]`) |
| `collapsed` | boolean | `true` | Allow collapsed reference (`[text][]`) |
| `shortcut` | boolean | `true` | Allow shortcut reference (`[text]`) |

```json
{
  "MD054": {
    "autolink": true,
    "inline": true,
    "full": true,
    "collapsed": false,
    "shortcut": false
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD054 can convert some link styles to allowed ones. Not all conversions are possible, making this rule partially fixable.

## Related Rules

- [MD052](md052.md) - Reference links should use defined labels
- [MD053](md053.md) - Reference definitions should be needed

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md054.md)
