# MD059 - emphasis-marker-style-math

Emphasis marker style should not conflict with math syntax.

**Tags:** emphasis, math

**Aliases:** emphasis-markers

**Fixable:** Yes (auto-fix available)

## Rationale

When using math syntax (e.g., LaTeX with dollar signs), emphasis markers can conflict with math delimiters. This rule helps ensure emphasis markers don't create ambiguity.

## Examples

### Incorrect

```markdown
The value is _x = 5_ which means...
```

May conflict with math syntax in some parsers.

### Correct

```markdown
The value is *x = 5* which means...
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD059 converts potentially conflicting emphasis markers.

## Related Rules

- [MD049](md049.md) - Emphasis style
- [KMD007](kmd007.md) - Math block delimiters
- [KMD011](kmd011.md) - Inline math balanced

## Additional Information

This rule is specific to mkdlint and is not part of the upstream markdownlint project.
