# MD053 - link-image-reference-definitions

Link and image reference definitions should be needed.

**Tags:** links, images

**Aliases:** link-image-reference-definitions

**Fixable:** Yes (auto-fix available)

## Rationale

Unused reference definitions (`[label]: url`) clutter the document source. If no link or image references the definition, it should be removed.

## Examples

### Incorrect

```markdown
Some content with no links.

[unused-label]: https://example.com
```

### Correct

```markdown
[Click here][my-link] for details.

[my-link]: https://example.com
```

## Configuration

| Option | Type | Default | Description |
|--------|------|---------|-------------|
| `ignored_definitions` | string array | `[]` | Reference definition labels to skip (e.g., `["//"]`) |

```json
{
  "MD053": {
    "ignored_definitions": ["//"]
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD053 removes unused reference definitions.

## Related Rules

- [MD052](md052.md) - Reference links and images should use defined labels

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md053.md)
