# MD056 - table-column-count

Table column count should be consistent.

**Tags:** table

**Aliases:** table-column-count

**Fixable:** No

## Rationale

All rows in a table should have the same number of columns. Mismatched column counts indicate a formatting error and may render incorrectly.

## Examples

### Incorrect

```markdown
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   |
```

The third row has only 1 column instead of 2.

### Correct

```markdown
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

This rule is not auto-fixable. Fixing column count mismatches requires understanding what content should be in the missing or extra cells.

## Related Rules

- [MD055](md055.md) - Table pipe style
- [MD058](md058.md) - Tables should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md056.md)
