# MD051 - link-fragments

Link fragments should be valid.

**Tags:** links

**Aliases:** link-fragments

**Fixable:** No

## Rationale

Link fragments (e.g., `#section-name`) should reference headings that actually exist in the document. Broken fragment links are invisible to readers until they try to click them.

## Examples

### Incorrect

```markdown
# Introduction

See [details](#nonexistent-section).
```

### Correct

```markdown
# Introduction

See [details](#introduction).
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

This rule is not auto-fixable. Fixing broken fragment links requires understanding the intended target heading.

## Related Rules

- [MD042](md042.md) - No empty links
- [MD052](md052.md) - Reference links and images should use defined labels

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md051.md)
