# MD055 - table-pipe-style

Table pipe style should be consistent.

**Tags:** table

**Aliases:** table-pipe-style

**Fixable:** Yes (auto-fix available)

## Rationale

Markdown tables can optionally have leading and trailing pipe characters. Using a consistent style throughout the document improves readability.

## Examples

### Incorrect (mixed styles)

```markdown
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |

Header 1 | Header 2
-------- | --------
Cell 1   | Cell 2
```

### Correct (consistent leading/trailing pipes)

```markdown
| Header 1 | Header 2 |
| -------- | -------- |
| Cell 1   | Cell 2   |
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD055 adjusts table rows to use consistent pipe style.

## Related Rules

- [MD056](md056.md) - Table column count
- [MD058](md058.md) - Tables should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md055.md)
