# MD052 - reference-links-images

Reference links and images should use a label that is defined.

**Tags:** links, images

**Aliases:** reference-links-images

**Fixable:** Yes (auto-fix available)

## Rationale

Reference-style links (`[text][label]`) and images (`\![alt][label]`) must have a matching definition (`[label]: url`). Undefined labels result in broken links.

## Examples

### Incorrect

```markdown
[Click here][undefined-ref]
```

### Correct

```markdown
[Click here][my-link]

[my-link]: https://example.com
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD052 adds stub reference definitions for undefined labels.

## Related Rules

- [MD051](md051.md) - Link fragments should be valid
- [MD053](md053.md) - Link and image reference definitions should be needed

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md052.md)
