# MD058 - blanks-around-tables

Tables should be surrounded by blank lines.

**Tags:** table, blank_lines

**Aliases:** blanks-around-tables

**Fixable:** Yes (auto-fix available)

## Rationale

Blank lines before and after tables improve readability and ensure proper rendering across parsers.

## Examples

### Incorrect

```markdown
Some text
| Header | Header |
| ------ | ------ |
| Cell   | Cell   |
More text
```

### Correct

```markdown
Some text

| Header | Header |
| ------ | ------ |
| Cell   | Cell   |

More text
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD058 inserts blank lines before and/or after tables where missing.

## Related Rules

- [MD055](md055.md) - Table pipe style
- [MD031](md031.md) - Fenced code blocks should be surrounded by blank lines

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md058.md)
