# MD050 - strong-style

Strong emphasis style should be consistent.

**Tags:** emphasis

**Aliases:** strong-style

**Fixable:** Yes (auto-fix available)

## Rationale

Strong emphasis can use asterisks (`**text**`) or underscores (`__text__`). Using a consistent style throughout the document improves readability.

## Examples

### Incorrect

```markdown
This is **bold** and this is __also bold__.
```

### Correct

```markdown
This is **bold** and this is **also bold**.
```

## Configuration

| Option | Type | Default | Values |
|--------|------|---------|--------|
| `style` | string | `"consistent"` | `consistent`, `asterisk`, `underscore` |

```json
{
  "MD050": {
    "style": "asterisk"
  }
}
```

## Auto-fix Behavior

When `--fix` is used, MD050 converts strong emphasis markers to match the expected style.

## Related Rules

- [MD049](md049.md) - Emphasis style
- [MD037](md037.md) - Spaces inside emphasis markers

## Additional Information

- [Upstream markdownlint rule documentation](https://github.com/DavidAnson/markdownlint/blob/main/doc/md050.md)
