# MD060 - dollar-in-code-fence

Dollar signs used before commands in fenced code blocks without output.

**Tags:** code

**Aliases:** dollar-in-code-fence

**Fixable:** Yes (auto-fix available)

## Rationale

Similar to MD014, but specifically targets fenced code blocks. Dollar sign prefixes on shell commands inside fenced code blocks are unnecessary when no output is shown, and they make copy-pasting commands harder.

## Examples

### Incorrect

```markdown
```bash
$ echo "Hello"
$ ls -la
```
```

### Correct

```markdown
```bash
echo "Hello"
ls -la
```
```

## Configuration

No configuration options for this rule.

## Auto-fix Behavior

When `--fix` is used, MD060 removes the `$` prefix from command lines in fenced code blocks.

## Related Rules

- [MD014](md014.md) - Dollar signs used before commands without showing output

## Additional Information

This rule is specific to mkdlint and is not part of the upstream markdownlint project.
